embark/uploader/boundedExecutor.py

```diff
@@ -50,32 +57,59 @@ def run_shell_cmd(self, cmd):
 
             # success
             logger.info(f"Success: {cmd}")
-            # TODO: inform asgi to propagate success to frontend
+
+            # take care of cleanup
+            if active_analyzer_dir:
+                shutil.rmtree(active_analyzer_dir)
+
+            # finalize db entry
+            if primary_key:
+                firmware = Firmware.objects.get(pk=primary_key)
+                firmware.end_date = datetime.now()
+                firmware.finished = True
+                firmware.save()
+
+            # success
+            logger.info(f"Successful cleaned up: {cmd}")
 
         except Exception as ex:
             logger.error(f"{ex}")
-            # TODO: inform asgi to propagate error to frontend
 
-    """
+            # take care of cleanup
```

Comment:
a little bit of duplicated code in regards to the successful case above. maybe refactor later

Reply:
Done